core/schema/directory.go

```diff
@@ -216,8 +216,43 @@ type subdirectoryArgs struct {
 	Path string
 }
 
-func (s *directorySchema) subdirectory(ctx context.Context, parent *core.Directory, args subdirectoryArgs) (*core.Directory, error) {
-	return parent.Directory(ctx, args.Path)
+func (s *directorySchema) subdirectory(
+	ctx context.Context,
+	parent dagql.ObjectResult[*core.Directory],
+	args subdirectoryArgs,
+) (res dagql.ObjectResult[*core.Directory], err error) {
+	query, err := core.CurrentQuery(ctx)
+	if err != nil {
+		return res, err
+	}
+	srv, err := query.Server.Server(ctx)
+	if err != nil {
+		return res, err
+	}
+	bk, err := query.Buildkit(ctx)
+	if err != nil {
+		return res, nil
+	}
+
+	dir, err := parent.Self().Directory(ctx, args.Path)
+	if err != nil {
+		return res, err
+	}
+
+	res, err = dagql.NewObjectResultForCurrentID(ctx, srv, dir)
+	if err != nil {
+		return res, err
+	}
+	if parent.Self().IsContentHashed {
+		res, err = core.MakeDirectoryContentHashed(ctx, bk, res)
```

Comment:
Mildly confused by this logic - the idea is that we keep a directory content-hashed upon selections? Not sure I understand what this is doing in terms of the behavior -- is it making a certain test pass?

Also would be nice if `MakeDirectoryContentHashed` set `IsContentHashed`.